### PR TITLE
fix bugs introduced by compat changes

### LINKED ISSUE #210: objects of type Cairo.#destroy cannot be finalized
```
julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.1 (2017-10-24 22:15 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-apple-darwin14.5.0

julia> Pkg.test("Cairo")
INFO: Computing test dependencies for Cairo...
INFO: Installing BaseTestNext v0.2.2
INFO: Testing Cairo
Image Surface  : Error During Test
  Got an exception of type ErrorException outside of a @test
  objects of type Cairo.#destroy cannot be finalized
  Stacktrace:
   [1] finalizer at ./base.jl:127 [inlined]
   [2] Type at /Users/monty/.julia/v0.6/Cairo/src/Cairo.jl:131 [inlined]
   [3] Type at /Users/monty/.julia/v0.6/Cairo/src/Cairo.jl:149 [inlined]
   [4] CairoImageSurface(::Int64, ::Int64, ::Int32) at /Users/monty/.julia/v0.6/Cairo/src/Cairo.jl:193
   [5] macro expansion at /Users/monty/.julia/v0.6/Cairo/test/runtests.jl:18 [inlined]
   [6] macro expansion at ./test.jl:860 [inlined]
   [7] anonymous at ./<missing>:?
   [8] include_from_node1(::String) at ./loading.jl:576
   [9] include(::String) at ./sysimg.jl:14
   [10] process_options(::Base.JLOptions) at ./client.jl:305
   [11] _start() at ./client.jl:371
Test Summary:   | Error  Total
Image Surface   |     1      1
ERROR: LoadError: Some tests did not pass: 0 passed, 0 failed, 1 errored, 0 broken.
while loading /Users/monty/.julia/v0.6/Cairo/test/runtests.jl, in expression starting on line 16
======================================================================[ ERROR: Cairo ]======================================================================

failed process: Process(`/Applications/Julia-0.6.app/Contents/Resources/julia/bin/julia -Ccore2 -J/Applications/Julia-0.6.app/Contents/Resources/julia/lib/julia/sys.dylib --compile=yes --depwarn=yes --check-bounds=yes --code-coverage=none --color=yes --compilecache=yes /Users/monty/.julia/v0.6/Cairo/test/runtests.jl`, ProcessExited(1)) [1]

============================================================================================================================================================
INFO: Removing BaseTestNext v0.2.2
ERROR: Cairo had test errors
```

```
julia> versioninfo()
Julia Version 0.6.1
Commit 0d7248e (2017-10-24 22:15 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin14.5.0)
  CPU: Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, skylake)
```

### PR DESCRIPTION
**Pull request by SimonDanisch**
fix #210

### CODE REVIEW

**Comment by lobingera** [comment]
Nice. How did you find this so quickly?

**Comment by SimonDanisch** [comment]
well, I was very sure that there must be a simple solution ;) and if you look at the code of `finalizer` (e.g. with `@edit`/`@which`) you can see that it's impossible, that `destroy` can end up in the error message that is getting displayed! So somewhere something must have switched the argument order.

File: src/Cairo.jl [hunk 1]
```diff
@@ -164,20 +165,20 @@ type CairoSurface{T<:Union{UInt32,RGB24,ARGB32}} <: GraphicsDevice
 
     @compat function (::Type{CairoSurface{T}}){T}(ptr::Ptr{Void}, w, h)
         self = new{T}(ptr, w, h)
-        finalizer(self, destroy)
+        finalizer(destroy, self)
```
Comment (fredrikekre) [inline src/Cairo.jl]: `@compat finalizer(destroy, self)`, see https://github.com/JuliaLang/Compat.jl/pull/422

**Comment by timholy** [comment]
Does this need a new tag?

**Comment by tkelman** [comment]
It needs fixing to use `@compat` as @fredrikekre reviewed, first

**Comment by fredrikekre** [comment]
I think I was mistaken -- `@compat` is recursive and in this case the whole function is `@compat`ed.

**Comment by lobingera** [comment]
@timholy I planned with tagging tomorrow. This here + compat.textwidth + taking 0.4 and win32 testing out.

File: src/Cairo.jl (hunk 1 above)
Comment (tkelman) [reply src/Cairo.jl]: @lobingera this should be fixed before the package os tagged

**Pull request merged**